onnxruntime/core/providers/cuda/nn/pool.cc

```diff
@@ -151,12 +152,17 @@ Status Pool<T, PoolType, NHWC>::ComputeInternal(OpKernelContext* context) const
   const Tensor* X = context->Input<Tensor>(0);
   const TensorShape& x_shape = X->Shape();
   const auto x_dims = x_shape.GetDims();
+  auto kernel_shape = pool_attrs_.kernel_shape;
 
-  if (x_shape.NumDimensions() < 3) {
+  if constexpr (NHWC) {
+    if (kernel_shape.size() < 2)
```

Comment:
Looks like lines 166 - 173 may be needed before we check `kernel_shape` size - that is the issue you are hitting with 2-D GlobalPool returning wronmg error that we don't support 1-D NHWC Pool

Reply:
May change the check to `kernel_shape.size() == 1` to avoid rejecting nhwc global pool